### PR TITLE
remove complex blob cache GC code

### PR DESCRIPTION
**Pull request by changweige**
Containerd can contorl if a committed snapshot should be removed. We don't have to consider if snapshot belongs to multiple images.

Just respond to containerd a committed snapshot shoud be removed.

Signed-off-by: Changwei Ge <gechangwei@bytedance.com>

### CODE REVIEW

File: pkg/converter/convert_unix.go [hunk 1]
```diff
@@ -832,7 +824,6 @@ func MergeLayers(ctx context.Context, cs content.Store, descs []ocispec.Descript
 			// Use this annotation to identify nydus bootstrap layer.
 			LayerAnnotationNydusBootstrap: "true",
 			// Track all blob digests for nydus snapshotter.
```
Comment (imeoer) [inline pkg/converter/convert_unix.go:826]: Should also remove the line comment.

File: pkg/cache/manager.go [hunk 2]
```diff
@@ -136,3 +86,24 @@ func (m *Manager) CacheUsage(ctx context.Context, blobID string) (snapshots.Usag
...
 	return usage, nil
 }
+
+func (m *Manager) RemoveBlobCache(blobID string) error {
+	blobCachePath := path.Join(m.cacheDir, blobID)
+	blobCacheSuffixedPath := path.Join(m.cacheDir, blobID+dataFileSuffix)
+	blobChunkMap := path.Join(m.cacheDir, blobID+chunkMapFileSuffix)
+	blobMeta := path.Join(m.cacheDir, blobID+metaFileSuffix)
+
+	stuffs := []string{blobCachePath, blobChunkMap, blobMeta, blobCacheSuffixedPath}
```
Comment (imeoer) [inline pkg/cache/manager.go:96]: Should remove `blobChunkMap` first, then `blobCachePath` or `blobCacheSuffixedPath`?

File: snapshot/snapshot.go [hunk 3]
```diff
@@ -423,34 +419,24 @@ func (o *snapshotter) Remove(ctx context.Context, key string) error {
...
-					log.G(ctx).WithError(err).Warn("failed to delete snapshot in cache manager")
-				}
-			} else {
-				log.G(ctx).WithError(err).Warnf("failed to get image ref from snapshot label %#v", snap.Labels)
-
-			}
-		}
+	if info.Kind == snapshots.KindCommitted {
+		blobDigest := info.Labels[label.CRILayerDigest]
+		go o.fs.RemoveBlobCache(blobDigest)
```
Comment (imeoer) [inline snapshot/snapshot.go:429]: Perhaps print a warning if failed here.

File: pkg/converter/convert_unix.go (hunk 1 above)
Comment (changweige) [reply pkg/converter/convert_unix.go]: removed

File: pkg/cache/manager.go (hunk 2 above)
Comment (changweige) [reply pkg/cache/manager.go]: Yes. Good catch. Fixed

File: snapshot/snapshot.go (hunk 3 above)
Comment (changweige) [reply snapshot/snapshot.go]: fixed

File: pkg/cache/manager.go [hunk 4]
```diff
@@ -136,3 +86,25 @@ func (m *Manager) CacheUsage(ctx context.Context, blobID string) (snapshots.Usag
...
+	blobMeta := path.Join(m.cacheDir, blobID+metaFileSuffix)
+
+	// NOTE: Delete chunk bitmap file before data blob
+	stuffs := []string{blobChunkMap, blobMeta, blobCachePath, blobCacheSuffixedPath}
+
+	for _, f := range stuffs {
+		err := os.Remove(f)
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				log.L.Warnf("file %s doest not exist.", f)
```
Comment (imeoer) [inline pkg/cache/manager.go:103]: For the new nydusd version, the `blobCachePath` always not exists, perhaps we just need to skip the error.

File: snapshot/snapshot.go [hunk 5]
```diff
@@ -423,41 +419,35 @@ func (o *snapshotter) Remove(ctx context.Context, key string) error {
...
-			if imageID, ok := snap.Labels[label.CRIImageRef]; ok {
-				if err := o.fs.DelSnapshot(imageID); err != nil {
-					log.G(ctx).WithError(err).Warn("failed to delete snapshot in cache manager")
-				}
-			} else {
-				log.G(ctx).WithError(err).Warnf("failed to get image ref from snapshot label %#v", snap.Labels)
-
+	if info.Kind == snapshots.KindCommitted {
+		blobDigest := info.Labels[label.CRILayerDigest]
+		go func(digest string) {
```
Comment (imeoer) [inline snapshot/snapshot.go:429]: The parameter seems unnecessary.

File: pkg/cache/manager.go (hunk 4 above)
Comment (changweige) [reply pkg/cache/manager.go:103]: fixed

File: snapshot/snapshot.go (hunk 5 above)
Comment (changweige) [reply snapshot/snapshot.go:429]: fixed

**Review by imeoer — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**